src/hotspot/share/compiler/directivesParser.cpp

```diff
@@ -318,7 +318,7 @@ bool DirectivesParser::set_option_flag(JSON_TYPE t, JSON_VAL* v, const key* opti
         char* s = NEW_C_HEAP_ARRAY(char, v->str.length+1,  mtCompiler);
         strncpy(s, v->str.start, v->str.length + 1);
         s[v->str.length] = '\0';
-        (set->*test)((void *)&s);
+        (set->*test)((void *)&s);  // Takes ownership.
```

Comment:
Would it be better to do the "set" only after validators have passed?

Reply:
Probably. Was just going with previous logic. I moved it to the end now. PTAL